docs/training/hyperparameter/search-methods/hp-custom.rst

```diff
@@ -0,0 +1,219 @@
+.. _topic-guides_hp-tuning-det_custom:
+
+#######################
+ Custom Search Methods
+#######################
+
++----------------------------------------------------------------+
+| API reference                                                  |
++================================================================+
+| :doc:`/reference/reference-searcher/custom-searcher-reference` |
++----------------------------------------------------------------+
+
+Determined supports defining your own hyperparameter search algorithms and provides search runner
+utilities for executing them.
+
+.. tip::
+
+   Remember that a Determined experiment is a set of trials, each corresponding to a point in the
+   hyperparameter space.
+
+To implement a custom hyperparameter tuning algorithm, subclass
+:class:`~determined.searcher.SearchMethod`, overriding its event handler methods. Note that
+overriding :meth:`~determined.searcher.SearchMethod.save_method_state` and
+:meth:`~determined.searcher.SearchMethod.load_method_state` of
+:class:`~determined.searcher.SearchMethod` is not mandatory. However, it is *recommended* as it
+enables fault tolerance.
+
+To run the custom hyperparameter tuning algorithm, you can use:
+
+-  :class:`~determined.searcher.LocalSearchRunner` that executes on your machine,
+-  :class:`~determined.searcher.RemoteSearchRunner` that runs on a Determined cluster.
+
+Both Search Runners execute the custom hyperparameter tuning algorithm and start a multi-trial
+experiment on a Determined cluster.
+
+The following sections explain the steps to take in order to implement and use a custom
+hyperparameter search algorithm. A detailed example can be found here:
+:download:`asha_search_method.tgz </examples/asha_search_method.tgz>`
+
+**********************************************
+ Experiment Configuration for Custom Searcher
+**********************************************
+
+You have to specify "custom" searcher type in the experiment configuration:
+
+.. code:: yaml
+
+   searcher:
+     name: custom
+     metric: validation_loss
+     smaller_is_better: true
+     unit: batches
+
+******************************
+ Search Method Implementation
+******************************
+
+Subclass :class:`~determined.searcher.SearchMethod`. Below is a starting template:
+
+.. code:: python
```

Comment:
I'm not crazy about duplicating our api definitions and documentation here in a code block.

Is there information in these comments which is different than what is in the docstrings of the actual interface?  If so, then it's either duplicate information, or the docstrings need improvement (at which point it will all be duplicated).

Also, there's no way to keep this up-to-date with the real interface.  However... the sphinx-generated docs for the real interface are already always up-to-date with the real interface.

Basically, I think the copy/paste benefit of this snippet is not worth the added complexity to our docs or our development process.